src/prog_models/utils/containers.py

```diff
@@ -35,66 +35,91 @@ def __init__(self, keys : list, data : Union[dict, np.array]):
                     [data[key]] if key in data else [None] for key in keys
                 ], dtype=np.float64)
         else:
-            raise ProgModelTypeError(f"Data must be a dictionary or numpy array, not {type(data)}")     
+            raise ProgModelTypeError(f"Data must be a dictionary or numpy array, not {type(data)}")
 
     def __reduce__(self):
         return (DictLikeMatrixWrapper, (self._keys, self.matrix))
 
-    def __getitem__(self, key : str) -> int:
-        row = self.matrix[self._keys.index(key)]
-        if len(row) == 1:
-            return self.matrix[self._keys.index(key)][0]
-        return self.matrix[self._keys.index(key)]
+    # get all values associated with a key, ex: all values of 'i'
+    def __getitem__(self, key: str) -> int:
+        row = self.matrix[self._keys.index(key)]  # creates list from a row of matrix
+        if len(row) == 1:  # if row list contains 1 value, returns that value (non-vectorized)
+            return row[0]
+        return row  # else returns entire row/list (vectorized case)
 
-    def __setitem__(self, key : str, value : int) -> None:
-        self.matrix[self._keys.index(key)] = np.atleast_1d(value)
+    # sets a row at the key given
+    def __setitem__(self, key: str, value: int) -> None:
+        index = self._keys.index(key)  # the int value index for the key given
+        self.matrix[index] = np.atleast_1d(value)
 
-    def __delitem__(self, key : str) -> None:
+    # removes row associated with key
+    def __delitem__(self, key: str) -> None:
         self.matrix = np.delete(self.matrix, self._keys.index(key), axis=0)
         self._keys.remove(key)
 
-    def __add__(self, other : "DictLikeMatrixWrapper") -> "DictLikeMatrixWrapper":
+    # add another matrix to the existing matrix
+    def __add__(self, other: "DictLikeMatrixWrapper") -> "DictLikeMatrixWrapper":
         return DictLikeMatrixWrapper(self._keys, self.matrix + other.matrix)
 
+    # creates iterator object for the list of keys
     def __iter__(self):
         return iter(self._keys)
 
+    # returns the length of key list
     def __len__(self) -> int:
         return len(self._keys)
 
-    def __eq__(self, other : "DictLikeMatrixWrapper") -> bool:
+    # checks if the other is a dictionary
```

Comment:
```suggestion
    # Compares two DictLikeMatrixWrappers (i.e. *Containers) or a DictLikeMatrixWrapper and a dictionary
```

It actually does more than what you're describing. It's checking if they're functionally equal. 

The logic in lines 75-79 compare the two if the other is a dictionary, while 80-82 compare if the other is another *Container